common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -355,6 +377,51 @@ int addHoverEvent(HoverEvent hoverEvent) {
         return hoverEvents.size();
     }
 
+    public int length() {
+        return parts.stream().mapToInt(StyledTextPart::length).sum();
+    }
+
+    public Optional<JsonObject> toJsonObject() {
+        String json = StringUtils.substringBeforeLast(getString(), "}") + "}"; // remove extra unnecessary info
+        try {
+            return Optional.of(JsonParser.parseString(json).getAsJsonObject());
+        } catch (JsonSyntaxException e) {
+            return Optional.empty();
+        }
+    }
+
+    public PartStyle getStyleAt(int index) {
+        for (StyledTextPart part : parts) if ((index -= part.length()) < 0) return part.getPartStyle();
+        return PartStyle.NONE;
+    }
+
+    public int getWidth() {
+        return parts.stream().mapToInt(StyledTextPart::width).sum();
+    }
+
+    public StyledText[] split(
+            String s) { // todo: doesn't work with color codes because the parts don't keep their color
+        return Arrays.stream(getString().split(s)).map(StyledText::fromString).toArray(StyledText[]::new);
+    }
+
+    public StyledText subtext(
+            int start, int end) { // todo: doesn't work with color codes because the parts don't keep their color
+        return StyledText.fromString(getString().substring(start, end));
+    }
+
+    public StyledText subtext(
+            int start) { // todo: doesn't work with color codes because the parts don't keep their color
+        return StyledText.fromString(getString().substring(start));
+    }
+
+    public StyledText replaceAll(String regex, String replacement) {
+        return StyledText.fromString(getString().replaceAll(regex, replacement));
+    }
+
+    public StyledText replace(String target, String replacement) {
+        return StyledText.fromString(getString().replace(target, replacement));
+    }
```

Comment:
Implement these in a way that you can only replace text parts, not formatting codes. You also need to do this for every part and construct a new styled text that way, because this way you lose some crucial info.